third_party/canvaskit/BUILD.gn

```diff
@@ -0,0 +1,36 @@
+# Copyright 2022 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import("//build/toolchain/wasm.gni")
+
+# This toolchain is only to be used by the canvaskit group below.
+wasm_toolchain("canvaskit") {
+  # Prevent accidental use of this toolchain.
+  assert(target_name == "canvaskit", "Only :canvaskit can use this toolchain.")
```

Comment:
I think `target_name` here is just the string passed to the `wasm_toolchain` template up on line 8, so this `assert` is not terribly helpful. I'd suggest removing it.

I've poked around the GN docs and source. It looks like it is not possible to set the visibility of a toolchain, or of a template. Also it looks like the label of the target that the toolchain is building is only available in the `tool()` definitions inside the toolchain, and only as string replacements `{{...}}`, not as variables that an assert could reference. So I think we're a little stuck, and that the comments that are already here are the best that can be done.

Reply:
> I think `target_name` here is just the string passed to the `wasm_toolchain` template up on line 8, so this `assert` is not terribly helpful. I'd suggest removing it.

Oh thanks for pointing this out. I'll remove the assert. I also think I should rename this to `canvaskit_toolchain` so it doesn't get confused with the `canvaskit` group.

> I've poked around the GN docs and source. It looks like it is not possible to set the visibility of a toolchain, or of a template.

I did the same and even tried it (in case it was an oversight in the docs) but gn yelled at me lol. That's when I changed it to an assert.

> So I think we're a little stuck, and that the comments that are already here are the best that can be done.

😞

Reply:
Ugh.. if I change this to `wasm_toolchain("canvaskit_toolchain") {` the output's subfolder also changes to `out/wasm_release/canvaskit_toolchain`.

@zanderso Do you know how I can control the output directory for the toolchain? I couldn't find any documentation on it.

Reply:
No, I don't know of any way to override the name there.